ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/manage/[ingestKeys]/[keyID]/page.tsx

```diff
@@ -38,29 +48,30 @@ type KeyDetailsProps = {
   };
 };
 
-export const runtime = 'nodejs';
+export default function Keys({ params: { ingestKeys, keyID } }: KeyDetailsProps) {
+  const [isDeleteKeyModalVisible, setIsDeleteKeyModalVisible] = useState(false);
+  const [isEditKeyNameModalVisible, setIsEditKeyNameModalVisible] = useState(false);
 
-export default async function Keys({
-  params: { environmentSlug, ingestKeys, keyID },
-}: KeyDetailsProps) {
-  const environment = await getEnvironment({
-    environmentSlug: environmentSlug,
-  });
+  const environment = useEnvironment();
 
-  const response = await graphqlAPI.request(GetKeyDocument, {
-    environmentID: environment.id,
-    keyID,
+  const { data, isLoading, error } = useGraphQLQuery({
```

Comment:
```suggestion
  const keyRes = useGraphQLQuery({
```

Destructuring removes the benefit of type narrowing. If you keep it as an object then handling `keyRes.error` and `keyRes.isLoading` will leave you with a non-optional `keyRes.data`. That way you don't have to handle the "missing `data` case

Reply:
I kept the same code you changed here: https://github.com/inngest/inngest/pull/964/files, just moved the location. I'm fine with changing the pattern, even though I find that new code more difficult to read. I don't think that's a blocker for the PR though

Reply:
Didn't know it was moved -- I'll approve the PR ✅ 

Getting rid of the destructuring might be a little less readable in this case, but the type narrowing makes it worthwhile. Otherwise you have to write unnecessary code just to satisfy the type system, since `data` will never actually be `undefined` if `error` and `isLoading` are both falsy